### PR TITLE
FIX: adjust relative uri references

### PR DESCRIPTION
**Pull request by mmcky**
This fix adjust relative `uri` references that are using `../` to produce corrected paths when compiling notebooks.

### CODE REVIEW

**Comment by arnavs** [comment]
@AakashGfude I'm having trouble interpreting the `diff` output from `nbdime`. Here's what I've got: 

* The build folders of immediately before and after we run the directory reorganization (with the two different Jupinxes). 
* An [adapted diff script](https://gist.github.com/arnavs/d4a4d0ca30a52db43c2bf58d93156e7a) which looks inside the reorganized (non-flat) structure for notebooks, and then (if it's not index) diffs it against the original.

Basically, we need to determine which of these changes are just coming from the changed links, or if this PR is breaking anything substantive. 

I'll send you the folders on slack.

**Comment by arnavs** [comment]
Note- I think the "same directory" links also need to be updated. i.e., if we link to `version_control` from `testing` (both of which are in `rst_files/more_julia`), then the generated link is: 

```
https://lectures.quantecon.org/jl/version_control.html
```

But it should be:

```
https://lectures.quantecon.org/jl/more_julia/version_control.html
```

**Comment by arnavs** [comment]
I can confirm that this doesn't introduce any outside errors into the notebooks. Execution is fine, as are includes of static files, equation numbering, etc. 

It's interesting, though, that markdown like: 

```
[continuous state Markov process](https://lectures.quantecon.org/jl/../tools_and_techniques/stationary_densities)
```

generates a "clickable" link like `https://lectures.quantecon.org/tools_and_techniques/stationary_densities`.

And shouldn't there be a `.html` at the end?
